common/persistence/sql/storage/mysql/eventsV2.go

```diff
@@ -29,24 +29,24 @@ import (
 const (
 	// below are templates for history_node table
 	addHistoryNodesQry = `INSERT INTO history_node (` +
-		`tree_id, branch_id, node_id, txn_id, data, data_encoding) ` +
-		`VALUES (:tree_id, :branch_id, :node_id, :txn_id, :data, :data_encoding) `
+		`shard_id, tree_id, branch_id, node_id, txn_id, data, data_encoding) ` +
+		`VALUES (:shard_id, :tree_id, :branch_id, :node_id, :txn_id, :data, :data_encoding) `
 
 	getHistoryNodesQry = `SELECT node_id, txn_id, data, data_encoding FROM history_node ` +
-		`WHERE tree_id = ? AND branch_id = ? AND node_id >= ? and node_id < ? ORDER BY tree_id, branch_id, node_id, txn_id LIMIT ? `
+		`WHERE shard_id = ? AND tree_id = ? AND branch_id = ? AND node_id >= ? and node_id < ? ORDER BY tree_id, branch_id, node_id, txn_id LIMIT ? `
```

Comment:
ORDER BY should also include shard_id and must be in the same order as primary key, so that we can get the benefit of primary key ordering.

Reply:
updated